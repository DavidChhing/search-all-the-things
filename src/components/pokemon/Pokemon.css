.loader {
  display: block;
  width: 400px;
  height: 300px;
  margin: auto;
}

label {
  border: 1px solid black;
  outline-style: solid;
  outline-width: thin;
  font-size: 20px;
  padding: 4px;
}

ol {
  font-size: 30px;
  margin: 2px;
  border: 1px solid green;
  width: 250px;
  bottom-border: 1px solid red;
}

li {
  border: 1px solid black;
  text-align: center;
  padding: 5px;
}
