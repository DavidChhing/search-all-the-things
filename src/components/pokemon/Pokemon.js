import React, { Component, Fragment } from 'react';
import { getPokemon } from '../../services/pokemonApi';
import styles from './Pokemon.css';
import Pageable from '../paging/Paging';

export default class Pokemon extends Component {
  state = {
    loading: false,
    currentPage: 1,
    totalPages: 1,
    effect: '',
    short_effect: '',
    pokemons: []
  };

  fetchPokemon = () => {
    const { currentPage, effect, short_effect } = this.state;
    this.setState({ loading: true, pokemons: [] }, () => {
      getPokemon(currentPage, { effect, short_effect })
        .then(res => {
          this.setState({
            totalPages: res.info.pages,
            pokemons: res.results,
            loading: false
          })
        });
    });
  }

  render() {
    const { loading, pokemons, effect, short_effect, currentPage, totalPages } = this.state;

    const pokemonComponents = pokemons.map(pokemon => {
      return <Pokemon key={pokemon.id}
      name={pokemon.name} />;
    });

    return (
      <Fragment>
        <select name="status" >

        </select>
    )

  }





}
