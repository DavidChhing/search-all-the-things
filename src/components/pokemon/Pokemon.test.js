import React from 'react';
import { shallow } from 'enzyme';
import Pokemon from './Pokemon';

describe('Pokemon test', () => {

  it('renders a pokemon', () )

});
