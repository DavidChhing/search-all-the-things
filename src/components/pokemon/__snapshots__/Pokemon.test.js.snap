// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon test renders a list of Pokemon 1`] = `
<React.Fragment>
  <label
    className="label"
  >
    Select Abilities
  </label>
  <select
    defaultValue=""
    name="ability"
    onChange={[Function]}
  />
  <Pageable
    currentPage={1}
    totalPages={1}
    updatePage={[Function]}
  />
  <div>
    <h1>
      Pokemon List
    </h1>
    <ol />
  </div>
</React.Fragment>
`;
